File: experiments/runs/2026-09-14_18-43-25_claim-office-example-mapping_external-tcr-kentbeck-2026-09-14-cc_opus-5-requesty-no-thinking/src/pricelist.ts
export interface ItemSpec {
  insuranceValue: number;
  basePremium: number;
}

const MAIN_ITEMS: Record<string, ItemSpec> = {
  sword: { insuranceValue: 1000, basePremium: 100 },
  amulet: { insuranceValue: 600, basePremium: 60 },
  staff: { insuranceValue: 800, basePremium: 80 },
  potion: { insuranceValue: 400, basePremium: 40 },
};

const COMPONENTS = new Set(['rune', 'moonstone']);

export const COMPONENT_SPEC: ItemSpec = { insuranceValue: 250, basePremium: 25 };

export const BLOCK_SIZE = 3;
export const BLOCK_PREMIUM = 60;

export function isComponent(type: string): boolean {
  return COMPONENTS.has(type);
}

export function lookup(type: string): ItemSpec {
  const main = MAIN_ITEMS[type];
  if (main) return main;
  if (isComponent(type)) return COMPONENT_SPEC;
  throw new Error(`unknown item type: ${type}`);
}

File: experiments/runs/2026-09-20_07-28-11_claim-office-java-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/main/java/PriceList.java
import java.math.BigDecimal;
import java.util.Map;
import java.util.Set;

/** The MHPCO price list: insurance values and base premiums per item type. */
public final class PriceList {

    private static final Map<String, int[]> MAIN_ITEMS = Map.of(
            "sword", new int[] {1000, 100},
            "amulet", new int[] {600, 60},
            "staff", new int[] {800, 80},
            "potion", new int[] {400, 40});

    private static final Set<String> COMPONENTS = Set.of("rune", "moonstone");

    /** Base premium of a building block of three alike components. */
    public static final BigDecimal BLOCK_PREMIUM = new BigDecimal(60);

    /** Number of alike components that form a building block. */
    public static final int BLOCK_SIZE = 3;

    private PriceList() {
    }

    public static boolean isComponent(String type) {
        return COMPONENTS.contains(type);
    }

    public static boolean isKnown(String type) {
        return MAIN_ITEMS.containsKey(type) || isComponent(type);
    }

    public static BigDecimal insuranceValue(String type) {
        requireKnown(type);
        return new BigDecimal(isComponent(type) ? 250 : MAIN_ITEMS.get(type)[0]);
    }

    public static BigDecimal basePremium(String type) {
        requireKnown(type);
        return new BigDecimal(isComponent(type) ? 25 : MAIN_ITEMS.get(type)[1]);
    }

    private static void requireKnown(String type) {
        if (!isKnown(type)) {
            throw new ClaimOfficeException("unknown item type: " + type);
        }
    }
}
